src/www/ui/api/Helper/UploadHelper.php

```diff
@@ -587,6 +587,61 @@ private function getMainLicenses($dbManager, $uploadId, $groupId)
     return $mainLicenses;
   }
 
+  /**
+   * Get the copyright list for given upload scanned by provided agents
+   * @param integer $uploadId        Upload ID
+   * @return array Array containing `copyright` and
+   * `copyright` for each upload tree item
+   */
+  public function getUploadCopyrightList($uploadId)
+  {
+    global $container;
+    $restHelper = $container->get('helper.restHelper');
+    $uploadDao = $restHelper->getUploadDao();
+    $agentDao = $container->get('dao.agent');
+
+    $uploadTreeTableName = $uploadDao->getUploadtreeTableName($uploadId);
+    $parent = $uploadDao->getParentItemBounds($uploadId, $uploadTreeTableName);
+
+    $scanProx = new ScanJobProxy($agentDao, $uploadId);
+    /** @var UIExportList $copyrightListObj
+     * UIExportList object to get copyright
+     */
+
+    $copyrightListObj = $restHelper->getPlugin('export-list');
+    $copyrightList = $copyrightListObj->getCopyrights($uploadId,
+      $parent->getItemId(), $uploadTreeTableName, -1, '');
+    if (array_key_exists("warn", $copyrightList)) {
+      unset($copyrightList["warn"]);
+    }
+
+    $responseList = array();
+    foreach ($copyrightList as $copyFilepath) {
+      $copyrightContent = array();
+      foreach ($copyrightList as $copy) {
+        if (($copyFilepath['content'] == $copy['content']) === true) {
+            array_push($copyrightContent,$copy['filePath']);
```

Comment:
Note from [phpnet documentation](https://www.php.net/manual/en/function.array-push.php)
> Note: If you use array_push() to add one element to the array, it's better to use $array[] = because in that way there is no overhead of calling a function. 
```suggestion
            $copyrightContent[] = $copy['filePath'];
```

Reply:
resolved